packages/services/src/epcisEvents/transactionEvent.ts

```diff
@@ -0,0 +1,103 @@
+import { CredentialSubject, VerifiableCredential } from '@vckit/core-types';
+import { IService } from '../types/index.js';
+import { issueVC } from '../vckit.service.js';
+import { IContext } from './types.js';
+import { getIdentifierByObjectKeyPaths } from './helpers.js';
+import { uploadJson } from '../storage.service.js';
+import { generateUUID, splitStringByDash } from '../utils/helpers.js';
+import { registerLinkResolver, DLREventEnum, EPCISEventAction, EPCISEventDisposition } from '../linkResolver.service.js';
+import { validateContextObjectEvent } from './validateContext.js';
+
+export interface ITransactionEvent {
+  data: CredentialSubject;
+}
+
+export interface ITransactionData {
+  sender: string
+  receiver: string;
+  locationUrl: string;
+  industryType: string;
+  transaction: {
+    type: string;
+    identifier: string;
+    documentURL: string;
+  };
+  livestockIds: string[];
+}
+
+export const processTransactionEvent: IService = async (transactionEvent: ITransactionEvent, context: IContext): Promise<VerifiableCredential> => {
+  const validationResult = validateContextObjectEvent(context);
+  if (!validationResult.ok) {
+    throw new Error(validationResult.value);
+  }
+
+  const { sender, receiver, locationUrl, industryType, transaction, livestockIds = [] } = transactionEvent.data as ITransactionData;
```

Comment:
Should not hardcode `{ sender, receiver, locationUrl, industryType, transaction, livestockIds = [] }`